packages/vscode-extension/src/debugging/DebugSession.ts

```diff
@@ -40,6 +40,20 @@ export class DebugSession implements Disposable {
     this.debugEventsListener.dispose();
   }
 
+  public async reconnectIfNeeded() {
+    const possibleRuntimes = await this.metro.fetchRuntimeList();
+    const availableAddresses =
+      possibleRuntimes?.map((runtime) => runtime.webSocketDebuggerUrl) ?? [];
+
+    if (!availableAddresses.includes(this.vscSession?.configuration?.websocketAddress)) {
```

Comment:
this seems overly complicated I think you just want to find runtime with the given ws URL, so this should be something like:
```
const currentWsAddress = this.vscSession?.configuration?.websocketAddress;
const hasCurrentWsAddress = possibleRuntimes?.some(runtime => runtime.webSocketDebuggerUrl === currentWsAddress)
```